### DISCUSSION TITLE
Discuss Metrics and Progress (planning and measurement viewpoints)

### DISCUSSION THREAD

**Discussion by haarer** [GfSE/SAF-Specification#51] [Ideas]
SAF makes the approach to structure MBSE work in to plannable parts - the Viewpoints.
We have reasoning why a viewpoint should be used (the concerns) and this can be used to plan the MBSE effort.

The information "which viewpoints to model" is currently not captured in the model, we don't have a viewpoint for this. The VP should capture which viewpoints to model, and also an estimate about the amount of model elements. E.g. for the System Usecase VP an estimate about the number of unique System Use Cases, and unique System Context Elements could be recorded.

For users, it might be also helpful, if the "way through the viewpoints" could be specified. We don't have a viewpoint for this, either.

Of course the actual content in a project is dependent from the "Engineering Story" of the project.
See [Bärentango](https://github.com/GfSE/SAF/raw/main/presentations/Modell-basierter%20B%C3%A4rentango%20mit%20dem%20System%20Architecture%20Framework.pdf) for one of the engineering stories we told at TdSE.

Perhaps it would be a good idea to support closing this open loop, i.e. by enabling SAF users to measure, if things are going along the plan.

IMO, that would mean to
- count to what extent the planned viewpoints have been reached
- count to what amount the estimation of model elements has been reached.
- count to what amount  relationships have been established. (e.g. are all context elements covered in use cases)

This is not an elaborated concept- but i want to trigger the discussion.

What do you think ?
What are your experiences with model metrics, what did work, what didn't?

Please contribute to this metrics discussion, @GfSE/saf !

**Comment by clalitsc** [comment]


**Reply by haarer** [reply]
Das ist jetzt etwas off-topic, aber ich gehe trotzdem kurz darauf ein.
Sicherlich braucht man die SE-Artefakte == SAF_InformationItems als in und out von Unternehmensprozessen.
Der Zuschnitt  der SE-Artefakte und damit die Verteilung der VP auf Information Items hänge aber von diesen Prozessen ab.
Das muß man also im Unternehmen zuschneiden.
Für SAF kann man einen Vorschlag für die InfoItems machen die im SE-Handbuch erwähnt sind.
Jegliche weitere Diskussion darüber bitte in einem eigenen Thread fortsetzen..

Zurück zum Thema.
Die eigentliche Frage war, ob und wie man Fortschrittsmetriken in SAF integrieren sollte, und welche Erfahrungen dazu vorliegen.
